CONTRIBUTING.md

```diff
@@ -24,19 +24,79 @@ Please follow these steps when making pull requests (PRs):
     Please give the PR a descriptive name.
 
 1.  As you make progress on your PR, keep your branch up-to-date with the `master` branch which
-        may have been updated after starting your PR.
+        may have been updated *after* starting your PR.
     Your PR **MUST** be updated to reflect changes to `master` in order to be merged.
-    Use the following procedure for updating your branch:
+    Use the following procedure for updating your branch and are ready to commit your changes:
 
     <!--- language: lang-shell -->
 
-        # While on your PR branch <issue-branch>
+        ## While on your PR branch <issue-branch> hosted at <your-remote> repository:
         git commit
         git checkout master
-        git pull origin --rebase
+        git pull <illixr-remote> --rebase                           ## Line A
         git checkout <issue-branch>
-        git rebase master
-        git push origin <issue-branch> --force-with-lease
+        git rebase master                                           ## Line B
+        git push <your-remote> <issue-branch> --force-with-lease    ## Line C
+
+    If you want to update your branch while you are still making changes to your branch:
+
+    <!--- language: lang-shell -->
+
+        ## While on your PR branch <issue-branch> hosted at <your-remote> repository:
+        git stash
+        git checkout master
+        git pull <illixr-remote> --rebase                           ## Line A
+        git checkout <issue-branch>
+        git rebase master                                           ## Line B
+        git stash apply <stash-number>
+        ## Finish your changes
+        git commit
+        git push <your-remote> <issue-branch> --force-with-lease    ## Line C
+
```

Comment:
This process may be onerous for new contributors or non-team members. Suggestions for facilitating collaboration are welcome.

Reply:
Re "this is complicated for new devs": should we recommend git-aliases or scripts for: new feature branch, rebase && incrment checkpoint, and rebase feature branch onto updated feature branch?

The latter is a workflow operation you have not yet described. Suppose Alice and Bob are collaborating on the same feature. Alice writes `alice/issue-123.3`, Bob writes a different `bob/issue-123.3`. Alice updates the GitHub repo first. What does Bob do? He should rebase his work onto the new `origin/issue-123.3` and increment the branch checkpoint to `issue-123.4`.